src/main/java/ro/andreiciortea/yggdrasil/http/HttpTemplateHandler.java

```diff
@@ -79,18 +80,18 @@ public void handleInstantiateTemp(RoutingContext routingContext) {
       message.toJson(), handleReply(routingContext, HttpStatus.SC_OK));
   }
 
-  // TODO: add payload validation
-  public void handleTemplateExtended(RoutingContext routingContext) {
+  public void handleAction(RoutingContext routingContext) {
     String body = routingContext.getBodyAsString();
     String artifactId = routingContext.request().getParam("artid");
     String[] uriSplitted = routingContext.request().absoluteURI().split(artifactId);
     String entityIri = uriSplitted[0] + artifactId;
-    String activity = uriSplitted[1];
+    String attribute = uriSplitted[1];
+    HttpMethod httpMethod = routingContext.request().method();
 
-    EventBusMessage message = new EventBusMessage(EventBusMessage.MessageType.TEMPLATE_ACTIVITY)
+    EventBusMessage message = new EventBusMessage(EventBusMessage.MessageType.TEMPLATE_ACTIVITY, httpMethod)
```

Comment:
Please view my comment on EventBusMessage. The HTTP method here should rather be passed as a header.

Reply:
changed it.